### DISCUSSION TITLE
How is the quality rating of an app determined? The YES and NO declarations

### DISCUSSION THREAD

**Discussion by Polype01** [kinfoundation/kin-rfcs#162] [KRE 4.0 Discussion]
"The self-assessment questions as at the implementation date are proposed to be as follows:"

- Can your end-users see their Kin public wallet address within your app?
- Can your end-users copy their Kin public wallet address to their device’s clipboard?
- Are your end-users able to export the private key of their in-app Kin wallet address?
- Can end-users import or connect an external Kin wallet to be used as the primary wallet within your Application?
- Do you permit end-users to deposit Kin to their in-app wallet balance?
- Do you permit end-users to withdraw 100% of their Kin balance from the app? (If you only enable partial withdrawals or you impose a withdrawal fee you must answer no).
- Does your app contain a tutorial or introduction for first time end-users to learn more about Kin, with a visible written hyperlink to “www.kin.org”?
- Does your app integrate a fiat-to-crypto payment solution from Ramp, Simplex, Wyre or MoonPay which enables your end-users to purchase Kin?
- Does your app incorporate the Kin logo as a currency symbol in all places where Kin can be spent by your end-users?
 
If an app declared 6 YES and 3 NO at the beginning of its activity and later on during the year, some extra App enhancements make factually 8 YES and 1 NO, will that (positive) update be taken into account in the quality rating determination? If yes, how will it be concretely reflected? From what I read, we only have penalties covered

**Comment by hann228** [comment]
This is currently under development but the essential principles will be like this:

- Developers can update their responses at any time. 
- The overall quality rating for the app will be updated monthly based on the latest responses given.  

This will give the developers flexibility to make any update to the questions whenever needed,  but the impact to the KRE algorithm is slower so it can be managed or overseen operationally (to avoid gaming).

**Reply by Polype01** [reply]
Will there be extra YES/NO questions submitted to app ? Is the app final? Some YES/NO questions have more weight or importance compared to others (ex: allowing  moonpay/simplex integration > being able to copy paste public address within app.) is that ok?

**Reply by Polype01** [reply]
Shouldn't one of the YES/NO questions be: Does your app/site provide your users multiple ways to spend KIN ? 
Integrating the spend concept should ne one of the main prerequisites of a 'KIN quality app' imo

**Reply by hann228** [reply]
> Will there be extra YES/NO questions submitted to app ? Is the app final? Some YES/NO questions have more weight or importance compared to others (ex: allowing moonpay/simplex integration > being able to copy paste public address within app.) is that ok?

I'm not sure if I follow - the questions may change over time or be updated as ecosystem priorities change or based on developer/community feedback.

> Shouldn't one of the YES/NO questions be: Does your app/site provide your users multiple ways to spend KIN ? Integrating the spend concept should ne one of the main prerequisites of a 'KIN quality app' imo

It is not necessary to ask quality related questions when the KRE algorithm deals with the matter directly.  For example KRE 4.0 requires developers to have a spend use case, and the median balance, number of active users spending etc is all taken into account.  In this respect it is not necessary to add an additional quality layer to this component (in our view) as poor spend performance or spend amount will already be taken into account for the purposes of the daily reward.

**Reply by tannerphilp** [reply]
Having different weightings for different components of the quality score could be a good option to explore. I thought that was a good suggestion @Polype01 

Any thoughts on that @hann228 ?

**Reply by hann228** [reply]
At the moment - there are 9 questions proposed which are weighted to a 100 point score in the KRE logic.  We are considering weighting the buy-module question to be worth 20 points, so that the remainder are all worth 10 points.   (20 x 1 + 10 x 8 = 100).
